source/common/profiler/profiler.cc

```diff
@@ -68,11 +68,33 @@ bool Heap::stopProfiler() { return false; }
 namespace Envoy {
 namespace Profiler {
 
+static absl::optional<tcmalloc::MallocExtension::AllocationProfilingToken> alloc_profiler =
+    absl::nullopt;
+
 absl::StatusOr<std::string> TcmallocProfiler::tcmallocHeapProfile() {
   auto profile = tcmalloc::MallocExtension::SnapshotCurrent(tcmalloc::ProfileType::kHeap);
   return tcmalloc::Marshal(profile);
 }
 
+bool TcmallocProfiler::startAllocationProfile() {
+  if (alloc_profiler) {
```

Comment:
should this return an absl::Status that can turn into an http error message lke "profiling already started"?

Reply:
Yes, that makes sense.